### ISSUE TITLE
Frontend contract modal should wait until after indexing is complete

### ISSUE THREAD

**Issue by ryscheng** [hypercerts-org/hypercerts#551]
### Describe the feature you'd like to request

Right now there's a gap between when we finish waiting for transaction to complete and when it gets indexed.

For example:
1. user creates hypercert - it shows success because we only wait until transaction confirms.
2. We goto the dashboard and it's not there (because we don't wait for the graph to finish indexing).

### Describe the solution you'd like

Can we update our frontend logic to
1. resolve only after indexing has completed?
2. add waiting for indexing to finish as a step in progress reporting after waiting for transaction to confirm?

### Describe alternatives you've considered

We could just cache results on the local client and show them without waiting for the graph to finish index.
That leads to a slightly more complicated operation of reconciling our local cache with the graph. 
Doable, but doesn't seem worth it?

**Comment by ryscheng** [comment]
Here's another user story reported on Telegram
https://github.com/hypercerts-org/hypercerts/issues/549

**Comment by bitbeckers** [comment]
These sound like client side solutions to me, assuming 'indexing' refers to the Graph:

1.  resolve only after indexing has completed?
The transaction can be resolved but not indexed by the Graph. So the Graph should only reload when there are updates. We can expose the option for live queries in the Graph. 

But removing the -old- rendered components could be suspended until the `tx.wait(5)` is resolved. This methods wait on 5 confirmations.

3.   add waiting for indexing to finish as a step in progress reporting after waiting for transaction to confirm?

We have no control over the Graph and it can be flaky. This could put the user on hold longer than comfortable. 

One solution could be to optimistically render the updated that and overlay the new certs with a spinner or something.

The main reason to keep this client side is that blocking rendering/state changes in the SDK feels to opiniated. The simplest exchange is: 
```put in data -> get txResponse -> response can become txReceipt or fail on status 0```

The FE should be able to handle all of these stages

**Comment by ryscheng** [comment]
Ya fair enough, I think I agree with you on these points, thanks for sharing your thoughts!

I'll retitle this issue to update the frontend Dapp to do some waiting, since this is an actual UX bug right now

**Comment by bitbeckers** [comment]
@ryscheng with the 5 block confirmations from #817 I noticed that during my mints the page loaded with the new certs and the create button when applicable. We might have covered this UX issue for now. 

After some testing when all open PRs are in, we can review and hopefully close this one

**Comment by bitbeckers** [comment]
Closing for now
